requirements/dev.txt

```diff
@@ -1,7 +1,3 @@
-black[jupyter]==24.3.0
-flake8==7.0.0
-isort==5.13.2
 pre-commit
 pytest-cov
-pyupgrade==v3.15.0
```

Comment:
Since this PR just replaces flake8, I think that you need to put back usort, black & pyupgrade in the requirements?

Also, you forgot to add ruff.

Reply:
I have run pre-commit instead of individual tools, and the packages in pre-commit have their own declaration and versions, so these installed packages are not used once you run pre-commit